esphome/components/samsung/samsung.cpp

```diff
@@ -0,0 +1,325 @@
+#include "samsung.h"
+#include <cinttypes>
+#include "esphome/core/helpers.h"
+
+namespace esphome {
+namespace samsung {
+
+static const char *const TAG = "samsung.climate";
+
+void SamsungClimate::transmit_state() {
+  if (this->current_climate_mode_ != climate::ClimateMode::CLIMATE_MODE_OFF &&
+      this->mode == climate::ClimateMode::CLIMATE_MODE_OFF) {
+    this->last_known_mode_ = this->current_climate_mode_;
+    this->send_power_state_(false);
+    return;
+  }
+
+  if (this->current_climate_mode_ == climate::ClimateMode::CLIMATE_MODE_OFF &&
+      this->mode != climate::ClimateMode::CLIMATE_MODE_OFF) {
+    this->send_power_state_(true);
+  }
+
+  std::memcpy(this->protocol_.raw, K_RESET, K_SAMSUNG_AC_EXTENDED_STATE_LENGTH);
+
+  this->current_climate_mode_ = this->mode;
+
+  this->set_climate_mode_(this->mode);
+  this->set_temp_(this->target_temperature);
+  this->set_swing_mode_(this->swing_mode);
+  this->set_fan_(this->fan_mode.has_value() ? this->fan_mode.value() : climate::CLIMATE_FAN_AUTO);
+
+  this->send_();
+}
+
+bool SamsungClimate::on_receive(remote_base::RemoteReceiveData data) {
+  if (!data.expect_item(SAMSUNG_AIRCON1_HDR_MARK, SAMSUNG_AIRCON1_HDR_SPACE)) {
+    return false;
+  }
+
+  ESP_LOGD(TAG, "Received Samsung A/C message");
+  for (uint8_t i = 0; i < 14; i++) {
+    if (i == 7) {
+      if (data.expect_item(SAMSUNG_AIRCON1_BIT_MARK, SAMSUNG_AIRCON1_MSG_SPACE)) {
+        ESP_LOGV(TAG, "Received MSG_SPACE %" PRIu8, i);
+      } else {
+        ESP_LOGW(TAG, "Failed to receive MSG_SPACE %" PRIu8, i);
```

Comment:
```suggestion
        ESP_LOGW(TAG, "Failed to receive MSG_SPACE %" PRIu8, i);
        return false;
```
Discard the entire packet due to an error during reception.